include/vulkan/layer/vk_layer_settings.h

```diff
@@ -21,44 +21,178 @@
 
 #pragma once
 
-#include <string>
-#include <vector>
-
 #include <vulkan/vk_layer.h>
 #include <vulkan/vulkan.h>
 
-namespace vku {
+// VK_EXT_layer_settings
+//
+// Name String
+//   VK_EXT_layer_settings
+//
+// Extension Type
+//   Instance extension
+//
+// Revision
+//    1
+//
+// Extension and Version Dependencies
+//    Requires Vulkan 1.0
+//
+// Contact
+//    Christophe Riccio christophe-lunarg
+//
+// Description
+//    This extension provides the VkLayerSettingsEXT struct that can be included in the pNext chain of the
+//    VkInstanceCreateInfo structure passed as the pCreateInfo parameter of vkCreateInstance
+//        The structure contains an array of VkLayerSettingValueEXT structures that define layer specific settings
+//    The extension also provides the vkEnumerateInstanceLayerSettingsEXT and vkEnumerateInstanceLayerSettingOptionEXT
+//    commands, useful for enumerating all layer settings and their possible values, respectively.
+//
+// Note
+//    The VK_EXT_layer_settings extension subsumes all the functionality provided in the [VK_EXT_validation_flags] extension
+//    and the [VK_EXT_validation_features] extension.
+//
+// New Commands
+//    vkEnumerateInstanceLayerSettingsEXT
+//    vkEnumerateInstanceLayerSettingOptionEXT
+//
+// New Structures
+//    array_int32
+//    array_int64
+//    array_float
+//    array_bool
+//    array_char
+//    VkLayerSettingValueEXT
+//    VkLayerSettingDescriptionEXT
+//    VkLayerSettingOptionEXT
+//    Extending VkInstanceCreateInfo :
+//        VkLayerSettingsEXT
+//
+// New Unions
+//    VkLayerSettingValueDataEXT
+//
+// New Enums
+//    VkLayerSettingValueTypeEXT
+
+#define VK_EXT_layer_settings 1
+#define VK_EXT_LAYER_SETTINGS_SPEC_VERSION 1
+#define VK_EXT_LAYER_SETTINGS_EXTENSION_NAME "VK_EXT_layer_settings"
+
+// This extension is exclusively used by VVL, and is NOT intended as a deliverable.
+// The value of the VK_STRUCTURE_TYPE is arbitrary. The only requirement,
+// is that it must not conflict with existing sTypes.
+//
+// NOTE: VK_STRUCTURE_TYPE_MAX_ENUM - 1 is used by the intel driver.
```

Comment:
\+ @mark-lunarg since I don't remember the full context.

We would eventually want to make an official extension to get a proper sType. Right?

Reply:
Yes this extension will become official but first we decided to implement it and tests it, potentially iterate on its definition.